Content.Server/Administration/AdminVerbSystem.cs

```diff
@@ -0,0 +1,199 @@
+using Content.Server.Administration.Commands;
+using Content.Server.Administration.Managers;
+using Content.Server.Administration.UI;
+using Content.Server.Configurable;
+using Content.Server.Disposal.Tube.Components;
+using Content.Server.EUI;
+using Content.Server.Ghost.Roles;
+using Content.Server.Inventory.Components;
+using Content.Server.Mind.Commands;
+using Content.Server.Mind.Components;
+using Content.Server.Players;
+using Content.Server.Verbs;
+using Content.Shared.Administration;
+using Content.Shared.Chemistry.Components.SolutionManager;
+using Content.Shared.Interaction.Helpers;
+using Content.Shared.Notification.Managers;
+using Content.Shared.Verbs;
+using Robust.Server.Console;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Administration
+{
+    /// <summary>
+    ///     System to provide various global admin/debug verbs
+    /// </summary>
+    public class AdminVerbSystem : EntitySystem
+    {
+        [Dependency] private readonly IConGroupController _groupController = default!;
+        [Dependency] private readonly IAdminManager _adminManager = default!;
+        [Dependency] private readonly EuiManager _euiManager = default!;
+        [Dependency] private readonly GhostRoleSystem _ghostRoleSystem = default!;
+        [Dependency] private readonly VerbSystem _verbSystem = default!;
+        
+        public override void Initialize()
+        {
+            SubscribeLocalEvent<GetOtherVerbsEvent>(AddDebugVerbs);
+        }
+
+        private void AddDebugVerbs(GetOtherVerbsEvent args)
+        {
+            if (!args.User.TryGetComponent<ActorComponent>(out var actor))
+                return;
+
+            var player = actor.PlayerSession;
+
+            // Delete verb
+            if (_groupController.CanCommand(player, "deleteentity"))
```

Comment:
Currently the way that admin command checks were done was very inconsistent. They usually used either `CanCommand`, `CanAdminMenu`, or `HasAdminFlag()`. Which one(s) should be used?

Reply:
CanCommand makes sense to me here